### PR TITLE
a bunch of compiler fixes

### PR DESCRIPTION
**Pull request by johnspackman**
fixes a bug where a missing Manifest.json in compile.json directory breaks the compile;
stops running migrations for every compile;
adds application groups to the compile (so that a set of applications can be chosen for compilation from the command line)
fixes a bug where the sourcemapping URL is broken in some IDEs (eg VS code)
fixes a bug in resolving the Qooxdoo path, it now favours the qooxdoo that the `qx` command is in

### CODE REVIEW

**Review by hkollmann — APPROVED** [review]

**Review by derrell — APPROVED** [review]
Blind approval.

File: source/resource/qx/tool/schema/compile-1-0-0.json [hunk 1]
```diff
@@ -73,6 +73,20 @@
             "description": "An arbitrary, but unique, short name for the application. Should be filename and URL friendly - IE no spaces or special characters",
             "type": "string"
           },
+          "group": {
+            "description": "An arbitrary, but unique, short name for grouping applications. Should be filename and URL friendly - IE no spaces or special characters",
+            "anyOf": [
+              {
+                "type": "string"
```
Comment (oetiker) [inline source/resource/qx/tool/schema/compile-1-0-0.json:80]:
```suggestion
                "type": "string",
                "pattern": "^[-_a-zA-Z0-9]+$"
```

File: source/resource/qx/tool/schema/compile-1-0-0.json [hunk 2]
```diff
@@ -73,6 +73,20 @@
...
+              }, 
+              {
+                "type": "array",
+                "items": {
+                  "type": "string"
```
Comment (oetiker) [inline source/resource/qx/tool/schema/compile-1-0-0.json:85]:
```suggestion
                  "type": "string",
                  "pattern": "^[-_a-zA-Z0-9]+$"
```

**Review by oetiker — APPROVED** [review]

**Review by cboulanger — CHANGES_REQUESTED** [review]
Very godd addition. Could you add a short info to the documentation to this PR? Thank you.

**Review by level420 — APPROVED** [review]

**Comment by oetiker** [comment]
@cboulanger anything missing still ?

**Review by cboulanger — APPROVED** [review]

**Comment by cboulanger** [comment]
Thank you @johnspackman !

**Pull request merged**